pkg/collector/horizontalpodautoscaler.go

```diff
@@ -46,6 +46,22 @@ func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1al
 	}
 
 	if autoscalingVersion == autodetect.AutoscalingVersionV2Beta2 {
+		metrics := []autoscalingv2beta2.MetricSpec{}
+
+		if otelcol.Spec.Autoscaler != nil && otelcol.Spec.Autoscaler.TargetMemoryUtilization != nil {
```

Comment:
I've removed thise if clauses, but at this point TargetCPUUtilization will always be set, as the webhook will set it to default if it is not in the CR.

Reply:
> but at this point TargetCPUUtilization will always be set, as the webhook will set it to default if it is not in the CR.

Is this correct behavior?

Shall we allow users to use independently CPU, memory or both?

Reply:
@pavolloffay Good qestion.  :-)  If we had done this correctly from the beginning I'd say no.  But at some point we decided if MaxReplicas was set then we would automatically set TargetCPUUtilization to a default value if it was not in the CR.

I can implement this whatever way you think is correct.

Reply:
In case the operator supports only CPU metric then it makes sense to default it when HPA is used. But if there are multiple metrics users should have choice which one to use and the operator should default only when no user config is provided.